compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -117,6 +117,26 @@ pub fn predicate_obligations<'a, 'tcx>(
         ty::PredicateAtom::Projection(t) => {
             wf.compute_projection(t.projection_ty);
             wf.compute(t.ty.into());
+
+            // Check item bounds apply.
+            //
+            // This is relying on this predicate being used to normalize any
+            // bounds. For example, for the projection predicate `<T as
+            // Iterator>::Item = U` we will get a bound `<T as Iterator>::Item:
+            // Sized`, which then normalizes to `U: Sized`.
+            let bounds = infcx.tcx.explicit_item_bounds(t.projection_ty.item_def_id);
+            let bound_obligations: Vec<_> = bounds
+                .iter()
+                .map(|&(bound, span)| {
+                    traits::Obligation::new(
+                        wf.cause(traits::BindingObligation(t.projection_ty.item_def_id, span)),
+                        param_env,
+                        bound.subst(infcx.tcx, t.projection_ty.substs),
```

Comment:
So -- where I got stuck here was that I assumed we would need to normalize after substitution. Does that happen elsewhere?

(I still feel like (not  in this PR...) I'd like to explore having the "bounds" predicate return a `Binder<Predicate>`, with the binder being the "self" type.)